src/BlockTypesController.php

```diff
@@ -8,45 +8,70 @@
 use Automattic\WooCommerce\Blocks\Integrations\IntegrationRegistry;
 
 /**
- * Library class.
- * Initializes blocks in WordPress.
+ * BlockTypesController class.
  *
  * @internal
  */
-class Library {
+class BlockTypesController {
 
 	/**
-	 * Initialize block library features.
+	 * Instance of the asset API.
+	 *
+	 * @var AssetApi
+	 */
+	protected $asset_api;
+
+	/**
+	 * Instance of the asset data registry.
+	 *
+	 * @var AssetDataRegistry
+	 */
+	protected $asset_data_registry;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param AssetApi          $asset_api Instance of the asset API.
+	 * @param AssetDataRegistry $asset_data_registry Instance of the asset data registry.
+	 */
+	public function __construct( AssetApi $asset_api, AssetDataRegistry $asset_data_registry ) {
+		$this->asset_api           = $asset_api;
+		$this->asset_data_registry = $asset_data_registry;
+		$this->init();
+	}
+
+	/**
+	 * Initialize class features.
 	 */
-	public static function init() {
-		add_action( 'init', array( __CLASS__, 'register_blocks' ) );
-		add_action( 'init', array( __CLASS__, 'define_tables' ) );
+	protected function init() {
+		add_action( 'init', array( $this, 'register_blocks' ) );
```

Comment:
Note, we don't need to define tables any more now that the stock table is in core.